pc/request.cpp

```diff
@@ -295,6 +295,20 @@ str product::get_symbol()
   return sym;
 }
 
+str product::get_base()
+{
+  str base;
+  get_attr( attr_id( "base" ), base);
+  return base;
+}
+
+str product::get_quote()
+{
+  str quote;
+  get_attr( attr_id( "quote_currency" ), quote);
```

Comment:
Does this fail gracefully if the attribute does not exist?

Reply:
I'm copy pasting `get_symbol` here. 

That said, my reading of the code is that it does fail gracefully. If the attribute is not found, `attr_id` returns the id of 0 (which, confusingly, represents an invalid attribute id, not the first attribute). get_attr on that id should give you a string whose char* is null.